pyzmq 13.0.0 bundled zeromq doesn't compile on CentOS

The latest version of pyzmq (13.0.0) comes bundled with libzmq version 3.3.0 (says bundled/zeromq/src/platform.hpp). It contains somes differences from the stable 3.2.2 that seems prevents compilation under some circumstances that are listed here:
< #define HAVE_SYS_EVENTFD_H 1

> /\* #undef HAVE_SYS_EVENTFD_H _/
> < #define ZMQ_HAVE_EVENTFD 1
> /_ #undef ZMQ_HAVE_EVENTFD _/
> < #define ZMQ_HAVE_SOCK_CLOEXEC 1
> /_ #undef ZMQ_HAVE_SOCK_CLOEXEC */

They seems to have been introduced in this changeset:
https://github.com/zeromq/pyzmq/commit/dd8240939a7353a5f62a73306014b748d6080517

Error while compiling the bundled zeromq happens such as:
bundled/zeromq/src/ip.cpp:45:14: error: ‘SOCK_CLOEXEC’ was not declared in this scope

These errors happens on a CentOS 5 system using GCC 4.7.2. The compilation works on an up-to-date Archlinux system.

I will contact the zeromq mailing list regarding these issues since they are the authors of these changes. Still, on some systems pyzmq 13.0.0 refuses to build using easy_install or pip because of this.

--- a/setup.py
+++ b/setup.py
@@ -100,15 +100,24 @@ doing_bdist = any(arg.startswith('bdist') for arg in sys.argv[1:])
 # but always assign it to configure
 
 configure_idx = -1
+fetch_idx = -1
 for idx, arg in enumerate(list(sys.argv)):
+    # track index of configure and fetch_libzmq
     if arg == 'configure':
         configure_idx = idx
+    elif arg == 'fetch_libzmq':
+        fetch_idx = idx
+    
     if arg.startswith('--zmq='):
         sys.argv.pop(idx)
         if configure_idx < 0:
-            sys.argv.insert(1, 'configure')
-            configure_idx = 1
+            if fetch_idx < 0:
+                configure_idx = 1
+            else:
+                configure_idx = fetch_idx + 1
+            sys.argv.insert(configure_idx, 'configure')
         sys.argv.insert(configure_idx + 1, arg)
+        break
 
 #-----------------------------------------------------------------------------
 # Configuration (adapted from h5py: http://h5py.googlecode.com)

